### ISSUE TITLE
Another name for "Mark Color" ?

### ISSUE THREAD

**Issue by felipesanches** [trufont/trufont#80]
A user complained that "Mark Color" made he think it had something to do with diacritical marks. Perhaps we could come up with a better, less ambiguous name for this option.

**Comment by davelab6** [comment]
Set Color

**Comment by felipesanches** [comment]
or maybe Highlight Color ?

**Comment by adrientetar** [comment]
It's defined upstream so we won't change it I think. Also there are quite some words which use mark, which is a synonym for trace. Mark alone doesn't mean much.

**Comment by davelab6** [comment]
I'm okay to leave it as-is

**Comment by twardoch** [comment]
"Mark" was a name I introduced in FLS5 but in FLVI, I renamed it to "Flag" (to disambiguate it from diacritical marks). Since all other font editors follow FontLab in implementing this functionality, feel free to follow the terminology change as well. 

A.

**Comment by twardoch** [comment]
In other words, since this is a FontLab-inspired feature (which made it into RoboFab and later to RoboFont), if you consider the FontLab implementation "upstream", then "Flag" is the current upstream term :)

**Comment by adrientetar** [comment]
I was more hinting at the UFO specification and defcon. "public.markColor" is the attribute in question.

**Comment by adrientetar** [comment]
I'm not opposed to revisiting how we label it at this point but we would decouple from the UFO terminology.

**Comment by twardoch** [comment]
Yep, that's true. Lots of UFO terminology is derived from Robofab, which in turn is derived from FontLab (Studio 5 or earlier). Including things that, in retrospect, I find unfortunate, which is why I'm changing them. 

Another example is the "Preferred Fa

**Comment by twardoch** [comment]
...mily name. In UFO, it's called "openTypeNamePreferredFamilyName", which is the old name that was used in the OpenType spec. I changed this to "Typographic Family Name" in TransType 4 and FontLab VI, and that change got adopted in OpenType 1.7. 

So currently, UFO and defcon are using an obsolete term while FontLab VI and the OpenType spec are using the new name.

It's up to you to decide which one to consider "upstream" for the TruFont UI.

**Comment by adrientetar** [comment]
> Lots of UFO terminology is derived from Robofab, which in turn is derived from FontLab 

Khaled Hosny was also complaining that the kerning in UFO is derived from FLS and too simplified.

**Comment by twardoch** [comment]
Yeah. When it comes to Layout, UFO's model for storage goes little beyond TrueType layout. Only a small fraction of the OpenType Layout model is upstreamed (kerning groups), and the rest is offloaded to what essentially is an external unparsable file (FEA). There is no notion of kerning directionality or other types of positioning. I've never thought UFO is in any way universal. It's very much an intersection of selected tool capabilities and a subset of a version of a particular font format, and rather short-sighted in design. But that's a discussion completely out of scope of this issue, I think :)

**Comment by davelab6** [comment]
I see no harm in having the ui text be as clear as possible if that
diverges from the spec. The ui should favor users over specifications over
implementors? Or another way around? :)

**Comment by felipesanches** [comment]
I think the UI should prioritize clarity of concepts for the users, even if we have to diverge from the technical jargon from the specs in order to reach the desired good user experience.

**Comment by twardoch** [comment]
Oh Dave, sed is so 90s, perl with the "e" flag is so much cooler :)

**Comment by graphicore** [comment]
[offtopic] should I18N be on the todo list? (Because it could be a way to customize the word via a translation)

**Comment by adrientetar** [comment]
I'd rather we get the renames done in UFO4.

**Comment by davelab6** [comment]
> I'd rather we get the renames done in UFO4.

So spec > users? Seems wrong to me, but you're the dictator ;)

**Comment by adrientetar** [comment]
I don't want to do a change just yet, I need some time for reflection.

**Comment by twardoch** [comment]
I think it's unlikely that this will change in UFO. Terminological changes "in code" (both XML and Python) are not worth the effort, they'll break other people's scripts and introduce inconsistencies. But it's a fundamental question in how much TruF wants to keep in sync developer-facing terminology and structures (code, property names, formats) and their user-facing representations. 

At FontLab, I tend to be a bit liberal, because history has taught me that much terminology inside, say, the OpenType spec, has been devised somewhat "ad-hoc" by software engineers, a long time ago, without much reflection on the UX. The best example is the term "Subfamily" used in the `name` table. Virtually no UI exposes that term to users, but instead the term "Style" is mostly used. Not sure whether it's a good thing, but that's what we have now. 

Such internal terminology won't change often, just like, say, Unicode character names won't be changed even though some of them are objectively wrong, and even the Unicode makers agree. 

But I'm not recommending any specific changes to be done at TruFont. Just presenting my point of view and presenting some historical background which led to certain things getting adapted. Specifically, the term "mark" for the color flag, was introduced in old FontLab without much reasoning and was ad-hoc, supported by the fact that the English knowledge among FontLab makers was not very good at that time. :)

**Comment by felipesanches** [comment]
Thanks for these comments, Adam.
I also agree with everything you said here.

**Comment by twardoch** [comment]
Ps. I'm also not advocating complete terminological freestyle. I.e. things that aren't confusing or terribly bad should stay as they are. Again, at FontLab, I'm not changing things just because I want to change things and shuffle things around. I always carefully weigh pros and contras of a given change, and only change things if I have really compelling reasons for it. One of the rationales I follow is, indeed, trying to avoid ambiguity. Renaming "mark" to "flag" was done exactly because these days, "mark" is most used in the context of diacritical marks, and "mark" has its strict definition in OpenType. When FontLab introduced the color mark (now "flag"), OpenType was still in early stages, so we didn't think of this ambiguity. And I actually did consider "highlight" for some time when looking for a better term when we were starting FLVI, after "flag" hit me when I realized that this is how you call such thing in e-mail apps. 

These days, I try to find good terminology in parallel implementations. For example, I consider CSS to be a good source of inspiration in many cases. Since CSS has the "line-height" property, I want to use the term "Line height" in the font editor UX. And since the "em" measurement has gotten good adoption in CSS and many people understand it (for the very first time in history ;) ), I'm inclined to use things like "Units per em" or "Em size" rather than, say, "UPM size". Not everyone agrees with it, and I myself agree that it's difficult decisions. Long-time users are confused by change, some users who are familiar with the OT spec are confused if the terms don't map 1:1, but new users will be happy if they find concepts familiar to them from other contexts, which these days is often "Adobe-like" drawing apps *and* CSS. :)
